lesson21_test_task1.py

```diff
@@ -0,0 +1,52 @@
+"""
+Take your implementation of the context manager class from Task 1 and write tests for it. Try to cover as many
+use cases as you can, positive ones when a file exists and everything works as designed. And also, write tests when
+your class raises errors or you have errors in the runtime context suite.
+"""
+
+import pytest
+import os
+from lesson21_task1 import CustomOpen
+
+
+@pytest.fixture
+def temp_file():
+    """Temp file for testing"""
+    test_file = "test_file.txt"
+    with open(test_file, "w") as f:
+        f.write("Test message!")
+    yield test_file
+
+    # Clean up
+    if os.path.exists(test_file):
+        os.remove(test_file)
+
```

Comment:
Greatly done!
Frankly speaking, I am impressed by your code quality. 
It feels like a production-level code, it would be great if everyone would write such a good code.